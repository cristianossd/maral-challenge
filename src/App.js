import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';

import Leaderboard from './components/leaderboard';
import LeaderboardResources from './components/leaderboard-resources';
import Login from './components/login';
import GenerateTeams from './components/generate-teams';
import TeamsScore from './components/teams-score';
import EventsScore from './components/events-score';
import EventsRanking from './components/events-ranking';
import { AUTH_TOKEN } from './constants';

import './assets/App.css';

class App extends Component {
  render() {
    const authToken = localStorage.getItem(AUTH_TOKEN);

    return (
      <div>
        <Switch>
          <Route exact path={`${process.env.PUBLIC_URL}/`} component={Leaderboard} />
          <Route exact path={`${process.env.PUBLIC_URL}/login`} component={Login} />
          {authToken && <Route exact path={`${process.env.PUBLIC_URL}/resources`} component={LeaderboardResources} />}
          {authToken && <Route exact path={`${process.env.PUBLIC_URL}/teams/generate`} component={GenerateTeams} />}
          {authToken && <Route exact path={`${process.env.PUBLIC_URL}/teams/score`} component={TeamsScore} />}
          {authToken && <Route exact path={`${process.env.PUBLIC_URL}/events/score`} component={EventsScore} />}
          {authToken && <Route exact path={`${process.env.PUBLIC_URL}/events/ranking`} component={EventsRanking} />}
        </Switch>
      </div>
    );
  }
}

export default App;
