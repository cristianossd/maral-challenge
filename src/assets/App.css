.App-logo {
  text-align: center;
  margin: 70px 0px 0px;
  padding: 0px 20px;
}

.App-logo img {
  width: 250px;
  height: auto;
}

.App-description {
  text-transform: uppercase;
  text-align: center;
  margin: 20px 0px 0px;
  padding: 0px 20px;
}

.App-description h4 {
  margin: 0px 0px 5px;
  font-weight: 700;
  color: #FFA500;
}

.App-description p {
  font-weight: 300;
  margin: 0px 0px 20px;
}

.App-description .btn {
  color: #191919;
  font-weight: 600;
  background-color: #FFA500;
  border-color: #FFA500;
  box-shadow: none;
}

.App-description .btn:hover {
  background-color: #CC8400;
  border-color: #CC8400;
  box-shadow: none;
}

.App-description .btn:active,
.App-description .btn:visited,
.App-description .btn:clicked,
.App-description .btn:focus {
  background-color: #CC8400;
  border-color: #FFA500;
  box-shadow: none;
}

.App-forms-select {
  padding: 0 20px 0;
}

.App-forms-select select {
  text-align: center;
}

.App-messages {
  text-align: center;
  padding: 30px 20px 20px;
  font-size: 12px;
}

.App-messages strong {
  color: #FFA500;
}
