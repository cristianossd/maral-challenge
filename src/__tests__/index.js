import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { HttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

global.localStorage = new LocalStorageMock;
const httpLink = new HttpLink({ uri: 'http://endpoint' });
const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
});

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <ApolloProvider client={client}>
          <App />
        </ApolloProvider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render properly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('.App').length).toBe(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render two routes without authentication', () => {
    const wrapper = shallow(<App />);

    const routes = wrapper.find('Route');
    expect(routes.length).toBe(2);
    expect(routes.at(0).prop('path')).toBe('/');
    expect(routes.at(1).prop('path')).toBe('/login');
  });

  it('should render routes with authentication', () => {
    global.localStorage.setItem('auth-token', 'aif1n209amsd');
    const wrapper = shallow(<App />);

    const routes = wrapper.find('Route');
    expect(routes.length).toBe(7);
    expect(routes.at(0).prop('path')).toBe('/');
    expect(routes.at(1).prop('path')).toBe('/login');
    expect(routes.at(2).prop('path')).toBe('/resources');
    expect(routes.at(3).prop('path')).toBe('/teams/generate');
    expect(routes.at(4).prop('path')).toBe('/teams/score');
    expect(routes.at(5).prop('path')).toBe('/events/score');
    expect(routes.at(6).prop('path')).toBe('/events/ranking');
  });
});
