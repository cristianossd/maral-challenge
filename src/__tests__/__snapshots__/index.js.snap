// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render properly 1`] = `
<div
  className="App"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
  </Switch>
</div>
`;
